### PR TITLE
Refactor `<Hero />`

### PR DESCRIPTION
**Pull request by nandereck**
🎟️ [Asana Task]()
🔍 [Preview Link]()

---

## Description
This PR updates the `<Hero />` to match the updated WPL spec.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

**Comment by nandereck** [comment]
@alexcarpenter - mind reviewing this WIP? Specifically `hero/index.tsx` and `hero/style.module.css`. You can preview [here](https://react-components-git-nels-herocreate-component-hashicorp.vercel.app/components/hero).

File: packages/hero/docs.mdx [hunk 1]
```diff
@@ -1,38 +1,41 @@
...
+                title: 'View documentation',
+                href: '/docs',
+                type: 'standalone-link'
+              }
+            ]
+          }}
+          url={"https://hashicorp.wistia.com/medias/glovs5atuj"}
+          smallImage={'https://www.datocms-assets.com/2885/1669673752-graphic-small.svg'}
+          mediumImage={'https://www.datocms-assets.com/2885/1669673746-graphic-medium.svg'}
+          largeImage={'https://www.datocms-assets.com/2885/1669673756-graphic.svg'}
```
Comment (alexcarpenter) [inline packages/hero/docs.mdx:30]: 💭 what about using something like:

```
images={{
  small: 'https://www.datocms-assets.com/2885/1669673752-graphic-small.svg',
  medium: '...',
  large: '...'
}}
```

File: packages/hero/types.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+> &
+  WithOptional<InlineVideoProps, 'url'> &
+  WithOptional<EyebrowProps, 'theme'>
+
+export interface HeroProps extends ContentProps {
+  backgroundColor?: string
+  smallImage?: string
+  mediumImage?: string
+  largeImage?: string
+}
```
Comment (alexcarpenter) [inline packages/hero/types.ts:19]: Getting some type errors for missing text in usage when implementing on the homepage. Not sure where the text error would be coming from. Possibly the eyebrow? The eyebrow is optional right?

**Review by alexcarpenter — COMMENTED** [review]
Looking great so far!

**Pull request closed without merging**